server/storage/fs_localCache.go

```diff
@@ -0,0 +1,167 @@
+package storage
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/dgraph-io/ristretto"
+	logx "github.com/ije/gox/log"
+)
+
+type LocalCacheFS struct{}
+
+// WIP: don't think it needs a TTL with a MaxCost, just preparing the logic for it.
+const TTL time.Duration = 0 * time.Second //time.Duration(30) * time.Minute,
+
+func (fs *LocalCacheFS) Open(root string, log *logx.Logger, isDev bool) (FSConn, error) {
+
+	backing, err := OpenFS("local:"+root, log, isDev)
+	if err != nil {
+		return nil, err
+	}
+
+	remove := func(name string) {
+		os.Remove(path.Join(root, name))
+	}
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 1e7,     // number of keys to track frequency of (10M).
+		MaxCost:     1 << 30, // maximum cost of cache (1GB).
```

Comment:
the root string can be `/tmp?maxCost=10gb`, then we can control the cache

Reply:
Done